### PR TITLE
Simplify parent lookup with Directory.GetParent

### PR DESCRIPTION
**Pull request by tjarls**
# Changed
- Changed: (Performance) Optimize a directory method by using Directory.GetParent, reducing I/O overhead

### CODE REVIEW

File: API.Tests/Services/DirectoryServiceTests.cs [hunk 1]
```diff
@@ -505,7 +505,7 @@ public void CopyFilesToDirectory_ShouldMoveAllFiles()
             }
 
             var ds = new DirectoryService(Substitute.For<ILogger<DirectoryService>>(), fileSystem);
-            ds.CopyFilesToDirectory(new []{$"{testDirectory}file_{0}.zip", $"{testDirectory}file_{1}.zip"}, "/manga/output/");
+            ds.CopyFilesToDirectory(new[] { $"{testDirectory}file_{0}.zip", $"{testDirectory}file_{1}.zip" }, "/manga/output/");
```
Comment (majora2007) [inline API.Tests/Services/DirectoryServiceTests.cs:508]: Please leave the space between new and [].

File: API/Services/DirectoryService.cs [hunk 2]
```diff
@@ -554,31 +554,19 @@ public IEnumerable<string> GetAllDirectories(string folderPath)
...
-
-               return Parser.Parser.NormalizePath(FileSystem.FileInfo
-                   .FromFileName(fileOrFolder).Directory.Parent
-                   .FullName);
-           }
-           catch (Exception)
-           {
-               return string.Empty;
-           }
+            // TODO: Write Unit tests
```
Comment (majora2007) [inline API/Services/DirectoryService.cs:561]: Remove the TODO, because you've written unit tests

**Review by majora2007 — CHANGES_REQUESTED** [review]
Please take care of the items, then this looks good to merge.

File: API.Tests/Services/DirectoryServiceTests.cs (hunk 1 above)
Comment (tjarls) [reply API.Tests/Services/DirectoryServiceTests.cs:508]: I did not actually do that. VS did automatically. I'll try to reverse it though

**Review by majora2007 — APPROVED** [review]

**Pull request merged**